modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/endpointComponents/PathParameterConfiguration.tsx

```diff
@@ -0,0 +1,184 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Text} from '@clayui/core';
+import ClayForm from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import {sub} from 'frontend-js-web';
+import React, {
+	Dispatch,
+	SetStateAction,
+	useCallback,
+	useEffect,
+	useState,
+} from 'react';
+
+import {Select} from '../fieldComponents/Select';
+import {DEFAULT_LANGUAGE_ID} from '../utils/constants';
+import {fetchJSON, getAllItems} from '../utils/fetchUtil';
+import {removeLeadingForwardSlash} from '../utils/string';
+
+interface PathParameterConfigurationProps {
+	data: Partial<APIEndpointUIData>;
+	selectedResponseBodySchema: SelectOption | undefined;
+	setData: Dispatch<SetStateAction<Partial<APIEndpointUIData>>>;
+}
+
+export default function PathParameterConfiguration({
+	data,
+	selectedResponseBodySchema,
+	setData,
+}: PathParameterConfigurationProps) {
+	const [pathParameterOptions, setPathParameterOptions] = useState<
+		SelectOption[]
+	>([]);
+
+	const [selectedPathParameter, setSelectedPathParameter] = useState<
+		SelectOption
+	>();
+
+	useEffect(() => {
+		setSelectedPathParameter(
+			pathParameterOptions.find(
+				(option) => option.value === data.pathParameter
+			)
+		);
+	}, [pathParameterOptions, data.pathParameter]);
+
+	const getObjectFieldLabel = (objectField: ObjectDefinition): string => {
+		if (
+			typeof objectField.label[
+				DEFAULT_LANGUAGE_ID as keyof LocalizedValue<string>
+			] !== 'undefined'
+		) {
+			return objectField.label[
+				DEFAULT_LANGUAGE_ID as keyof LocalizedValue<string>
+			]!;
+		}
+		else if (Object.keys(objectField.label).length !== 0) {
+			return Object.values(objectField.label)[0];
+		}
+		else {
+			return objectField.name;
+		}
+	};
+
+	const handleSelectPathParameter = (value: string) => {
+		setData((previousValue) => ({
+			...previousValue,
+			pathParameter: value,
+		}));
+
+		setSelectedPathParameter(
+			pathParameterOptions.find((option) => option.value === value)
+		);
+	};
+
+	const fetchObjectDefinitionsByExternalReferenceCode = (
```

Comment:
Do you think that it could be `fetchFilteredObjectDefinitionsByExternalReferenceCode`?